cli/pkg/backend/aws/k8s_compute_backend.go

```diff
@@ -203,38 +206,128 @@ func (k8s *K8SComputeBackend) PrintLogs(ctx context.Context, stackName string, s
 	logrus.Infof("Found %d matching pods.", len(pods.Items))
 
 	for _, pod := range pods.Items {
-		logrus.Infof("... streaming logs from pod %s ...", pod.Name)
-
-		logs, err := k8s.ClientSet.CoreV1().Pods(k8s.HappyConfig.K8SConfig().Namespace).GetLogs(pod.Name, &corev1.PodLogOptions{
-			Follow: false,
-		}).Stream(ctx)
+		err = k8s.streamPodLogs(ctx, pod, false)
 		if err != nil {
-			return errors.Wrapf(err, "unable to retrieve logs from pod %s", pod.Name)
+			logrus.Error(err.Error())
 		}
-		defer logs.Close()
-		reader := bufio.NewScanner(logs)
+	}
+	return nil
+}
 
-		for reader.Scan() {
-			logrus.Info(string(reader.Bytes()))
-		}
-		logrus.Infof("... done streaming ...")
+func (k8s *K8SComputeBackend) streamPodLogs(ctx context.Context, pod corev1.Pod, follow bool) error {
+	logrus.Infof("... streaming logs from pod %s ...", pod.Name)
+
+	logs, err := k8s.ClientSet.CoreV1().Pods(k8s.HappyConfig.K8SConfig().Namespace).GetLogs(pod.Name, &corev1.PodLogOptions{
+		Follow: follow,
+	}).Stream(ctx)
+	if err != nil {
+		return errors.Wrapf(err, "unable to retrieve logs from pod %s", pod.Name)
 	}
+	defer logs.Close()
+	reader := bufio.NewScanner(logs)
+
+	for reader.Scan() {
+		logrus.Info(string(reader.Bytes()))
+	}
+	logrus.Infof("... done streaming ...")
 	return nil
 }
 
 func (k8s *K8SComputeBackend) RunTask(ctx context.Context, taskDefArn string, launchType config.LaunchType) error {
-	// TODO: not implemented
-	return errors.New("not implemented")
+	// Get the cronjob and create a job out of it
+	if len(taskDefArn) >= 2 {
+		if taskDefArn[0] == '"' && taskDefArn[len(taskDefArn)-1] == '"' {
+			taskDefArn = taskDefArn[1 : len(taskDefArn)-1]
+		}
+	}
+	cronJob, err := k8s.ClientSet.BatchV1().CronJobs(k8s.HappyConfig.K8SConfig().Namespace).Get(ctx, taskDefArn, v1.GetOptions{})
+	if err != nil {
+		return errors.Wrap(err, "unable to retrieve a template cronjob")
+	}
+
+	jobId := fmt.Sprintf("%s-%s-job", taskDefArn, uuid.NewUUID())
+	jobDef := k8s.createJobFromCronJob(cronJob, jobId)
+	jb, err := k8s.ClientSet.BatchV1().Jobs(k8s.HappyConfig.K8SConfig().Namespace).Create(ctx, jobDef, v1.CreateOptions{})
+	if err != nil {
+		return errors.Wrap(err, "unable to create a k8s job")
+	}
+
+	logrus.Info("Waiting for all the pods to start")
+	// Make sure pods have started
+	counter := 0
+	var pods *corev1.PodList
+	for {
+		time.Sleep(5 * time.Second)
```

Comment:
I think this helper would be a little nicer here: https://github.com/chanzuckerberg/happy/blob/558e9e003eb8312f8e2ea35234b81a031b9d27aa/cli/pkg/util/time.go#L24. It also gives you a timeout period so you don't run an infinite loop by accident.

Reply:
It reruns until either the timeout is hit or you return a nil error

Reply:
then you can remove the counter

Reply:
How do you successfully interrupt this when you have successfully accomplished what you needed before timeout?

Reply:
you return a nil error